### PR TITLE
Improper escape code in AuthorizeNetAIM::addLineItem

### PR DESCRIPTION
**Pull request by jdavidbakr**
Preg replace needed to escape the pipe character

### CODE REVIEW

**Comment by ashtru** [comment]
Thanks you for submitting the PR @jdavidbakr , we will review it soon.

File: lib/AuthorizeNetAIM.php [hunk 1]
```diff
@@ -269,12 +269,12 @@ public function addLineItem($item_id, $item_name, $item_description, $item_quant
         $line_item = "";
         $delimiter = "";
         foreach (func_get_args() as $key => $value) {
-            $line_item .= $delimiter . preg_replace('/|/', '_', $value);
+            $line_item .= $delimiter . preg_replace("/\|/", '_', $value);
```
Comment (gauthierm) [inline lib/AuthorizeNetAIM.php]: Use single quotes here so the `\` character is considered a regular expression escape character and not a string escape character.

Even better:

```php
$line_item .= $delimiter . str_replace('<|>', '_', $value);
```

The referenced PDF documentation in the original commit notes the delimiter sequence is `<|>` and not `|`. There is no reason to use a regular expression here when it's just doing a simple search/replace operation.

**Comment by gauthierm** [comment]
https://github.com/AuthorizeNet/sdk-php/pull/291 is a duplicate of this.

**Comment by ashtru** [comment]
Thanks for the PR, @jdavidbakr 
The changes are already merged as part of #312 .

**Comment by ashtru** [comment]
Duplicate of #291 . Closing.

**Pull request closed without merging**